### ISSUE TITLE
[data grid] no way to override GridPanelFooter

### ISSUE THREAD

**Issue by andytsoi1** [mui/mui-x#10909]
### Steps to reproduce

Link to live example: https://codesandbox.io/s/github/mui/mui-x/tree/master/templates/x-data-grid?file=/src/demo.tsx

Steps:
1. Go to column header's filter and press it to open the filter panel

### Current behavior

no way to override GirdPanelFooter

### Expected behavior

Should be able to add any button or element in to the GirdPanelFooter as i wanted with props or callback for applying the filter.

### Context

I am trying to add a button of "Submit" to the GirdPanelFooter, so the filter panel doesn't apply filter until i press the "Submit" button. Seems no way to do that now.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
System:
    OS: macOS 13.3
  Binaries:
    Node: 18.12.1 - /usr/local/bin/node
    Yarn: 3.3.1 - /usr/local/bin/yarn
    npm: 8.19.2 - /usr/local/bin/npm
  Browsers:
    Chrome: 119.0.6045.105
    Edge: Not Found
    Safari: 16.4
  npmPackages:
    @emotion/react: 11.11.1 => 11.11.1 
    @emotion/styled: 11.11.0 => 11.11.0 
    @mui/base:  5.0.0-beta.8 
    @mui/core-downloads-tracker:  5.14.9 
    @mui/material: 5.14.2 => 5.14.2 
    @mui/private-theming:  5.14.9 
    @mui/styled-engine:  5.14.9 
    @mui/system:  5.14.9 
    @mui/types:  7.2.4 
    @mui/utils:  5.14.9 
    @mui/x-data-grid:  6.10.2 
    @mui/x-data-grid-pro: 6.10.2 => 6.10.2 
    @mui/x-date-pickers: 6.10.2 => 6.10.2 
    @mui/x-license-pro:  6.10.2 
    @types/react: 18.2.17 => 18.2.17 
    react: 18.2.0 => 18.2.0 
    react-dom: 18.2.0 => 18.2.0 
    typescript: ~4.9.4 => 4.9.5 
```
</details>

**Search keywords**: datagrid GridPanelFooter

**Comment by romgrk** [comment]
Have you looked at the implementation of the filter panel? I don't think just adding a panel footer is enough, you'd need to reimplement the whole panel.

**Comment by andytsoi1** [comment]
but isn't it abit too much if i just want to add a button? But if that the only way, may i know is there any example on how to implement the whole filter panel with all the `GridPanelFooter` and `GridFilterForm` and where can i get all these props from? I don't see the `FilterPanel` Slot have all these props at all....
` <GridFilterForm
              key={item.id == null ? index : item.id}
              item={item}
              applyFilterChanges={applyFilter}
              deleteFilter={deleteFilter}
              hasMultipleFilters={hasMultipleFilters}
              showMultiFilterOperators={index > 0}
              multiFilterOperator={filterModel.logicOperator}
              disableMultiFilterOperator={index !== 1}
              applyMultiFilterOperatorChanges={applyFilterLogicOperator}
              focusElementRef={index === items.length - 1 ? lastFilterRef : null}
              logicOperators={logicOperators}
              columnsSort={columnsSort}
              {...filterFormProps}
            />`

**Comment by romgrk** [comment]
The problem isn't adding the button, it's disabling the existing logic. There is no example for that as it's an uncommon use-case. I would recommend to take a look at the grid source code directly, in particular `GridFilterPanel.tsx`. If you want to turn this in a feature request that's possible but we'll have to wait to see the popularity of the issue before prioritizing.

**Comment by andytsoi1** [comment]
i did look on the source code, but no props were pass into the filterPanel in the slot, but anyway i guess there is no straight forward way to solve this at the moment.
I do see some request for this feature of "apply filter" button : https://github.com/mui/mui-x/issues/3804 , but yea let me request this as a feature too.

**Comment by romgrk** [comment]
Closed in favor of #10938
